Marlin/pins_PRINTRBOARD.h

```diff
@@ -21,18 +21,156 @@
+ *  commented out.  In that case the X-STOP capability is probably lost.
+ *
+ *  The "Printrboard" extension is preferred because it contains a better bootloader.
+ */
+
+/**
+ * Rev B changes to pin functions
+ *
+ *  Since the SPI module is in MASTER mode, the SDSS signal is manually controlled by
+ *  by the software which means the SDSS function can be assigned to any convenient pin.
+ *  This allows the following:
+ *    SDSS pin number changes depending on where the active SD card is (on board, on LCD)
+ *    Y_STOP is always at the Y-stop connector
+ *
+ *  SD card detect is enabled for on board SD card
+ *
+ *  PANELOLU2 section
+ *    In Rev A the SDSS pin used the same pin as FILAMENT_WIDTH_SENSOR.  In Rev B the
+ *      FILWIDTH_PIN was moved to analog input A4 (when PANELOLU2 is enabled) .  That
+ *      makes it available on pin 8 (TCK) on the SV1 connector on Printrboard Adapter.
+ *      http://blog.think3dprint3d.com/2013/04/panelolu2-for-ramps-and-printrboard.html
+ *
+ *  VIKI section
+ *    SDSS moved to resolve conflict with BTN_ENC
+ *      This was caused by an error in the table in the VIKI 2 wiring diagram published
+ *      by Panucatt maker of VIKI).  It was moved to the pin shown in the graphic.
+ *    SD card detect added
+ *      Caused by another error in the wiring diagram table.  It was set to the pin shown
+ *      in the graphic.
+ *    RESET - added a comment in case the user would rather the red button on the panel be
+ *            be tied to the KILL_PIN instead of RESET.  RESET is shown in the graphic.
+ *
+ * None of these changes have been done to the Rev A section.
+ */
+
+/**
+ * Rev A refers to "FASTIO" pins, "Non-FastIO" pins and "Teensyduino" pins.
+ *   "FASTIO" pins are what the STANDARD boards use.  This is the FASTIO AT90USB646/1286 map
+ *      when the "AT90USBxx_TEENSYPP_ASSIGNMENTS" flag is commented out.  Rev B sometimes
+ *      calls this the STANDARD pin map.
+ *   "Non-FastIO" pins and "Teensyduino" pins are a subset of the pin map used by the
+ *      TEENSYPP boards.  Rev B sometimes calls this the TEENSYPP pin map.
  */
 
 #ifndef __AVR_AT90USB1286__
-  #error "Oops!  Make sure you have 'Teensy++ 2.0' selected from the 'Tools -> Boards' menu."
+  #error "Oops!  Make sure you have 'Teensy++ 2.0', "AT90USB1286_TEENSYPP", "AT90USB1286_STANDARD" or "Printrboard" selected from the 'Tools -> Boards' menu."
 #endif
 
-#if ENABLED(AT90USBxx_TEENSYPP_ASSIGNMENTS)  // use Teensyduino Teensy++2.0 pin assignments instead of Marlin traditional.
-  #error "These Printrboard assignments depend on traditional Marlin assignments, not AT90USBxx_TEENSYPP_ASSIGNMENTS in fastio.h"
+// determine which pin map to use
+#if defined(ARDUINO_AT90USB1286_TEENSYPP)
+  #define USE_TEENSYPP_PIN_MAP true // using TEENSYPP pin map
+#elif defined(ARDUINO_AT90USB1286_STANDARD)
+  #define USE_TEENSYPP_PIN_MAP false // using STANDARD pin map
+#else
+  #define USE_TEENSYPP_PIN_MAP true     // default to TEENSYPP pin map
```

Comment:
For the pin mapping stuff, first, please make a separate PR from this one and remove the pin mapping-related changes from this PR. Second, let's not include both pin mappings in a single file. Use these conditions to choose which pins file to include instead, then put the pins definitions into new files `pins_PRINTRBOARD-Teensy.h` and `pins_PRINTRBOARD-AT90USB.h`.